builtin/providers/google/resource_storage_bucket.go

```diff
@@ -64,14 +103,60 @@ func resourceStorageBucketCreate(d *schema.ResourceData, meta interface{}) error
 	log.Printf("[DEBUG] Created bucket %v at location %v\n\n", res.Name, res.SelfLink)
 
 	// Assign the bucket ID as the resource ID
+	d.Set("self_link", res.SelfLink)
 	d.SetId(res.Id)
 
 	return nil
 }
 
 func resourceStorageBucketUpdate(d *schema.ResourceData, meta interface{}) error {
-	// Only thing you can currently change is force_delete (all other properties have ForceNew)
-	// which is just terraform object state change, so nothing to do here
+	config := meta.(*Config)
+
+	// Get the bucket and acl
+	bucket := d.Get("name").(string)
+	acl := d.Get("predefined_acl").(string)
+	location := d.Get("location").(string)
+
+	// Create a bucket, setting the acl, location and name.
+	sb := &storage.Bucket{Name: bucket, Location: location}
+
+	if d.HasChange("website") {
+		if v, ok := d.GetOk("website"); ok {
+			websites := v.([]interface{})
+
+			if len(websites) > 1 {
+				return fmt.Errorf("At most one website block is allowed")
+			}
+
+			website := websites[0].(map[string]interface{})
+
+			if v, ok := website["website_not_found_page"]; ok {
+				sb.Website.NotFoundPage = v.(string)
+			} else {
+				sb.Website.NotFoundPage = ""
```

Comment:
I was being explicit since it seems like `patch` makes use of the `omitempty` flag, but I can remove these if you think they're redundant.